packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -81,9 +81,32 @@ class _KeyboardClientImpl implements mojom.KeyboardClient {
   }
 }
 
+/// Configurable state of an input field.
+class InputValue {
+  const InputValue({ this.text: '', this.selection });
+
+  /// The current text being edited.
+  final String text;
+
+  /// The range of text that is currently selected.
+  final TextSelection selection;
+
+  static const InputValue empty = const InputValue();
+
+  bool operator ==(Object other) {
```

Comment:
This was in the original proposal from #1586.  Removed - it looks like it's unnecessary.